e2e/cypress/integration/notifications/desktop_notifications_spec.js

```diff
@@ -270,6 +270,174 @@ describe('Desktop notifications', () => {
+                const messageWithNotification = `random message with mention @${user.username}`;
+
+                // Have another user send a post with a mention
+                cy.postMessageAs({sender: testUser, message: messageWithNotification, channelId: channel.id});
+
+                // * Desktop notification is not received
+                cy.get('@withNotification').should('not.have.been.called');
+
+                // # Have another user post a direct message
+                cy.apiCreateDirectChannel([user.id, testUser.id]).then(({channel: dmChannel}) => {
+                    cy.postMessageAs({sender: testUser, message: 'hi', channelId: dmChannel.id});
+                    ignoreUncaughtException();
+
+                    // # DM notification is not received
+                    cy.get('@withNotification').should('not.have.been.called');
+                });
+            });
+        });
+    });
+
+    it('MM-T496 - Channel notifications: Desktop Notification Sounds OFF', () => {
+        cy.apiCreateUser().then(({user}) => {
+            cy.apiAddUserToTeam(testTeam.id, user.id);
+            cy.apiLogin(user);
+            ignoreUncaughtException();
+
+            // Visit town-square.
+            cy.visit(`/${testTeam.name}/channels/town-square`);
+            spyNotificationAs('withNotification', 'granted');
+
+            // # Click hamburger main menu.
+            cy.get('#sidebarHeaderDropdownButton').click();
+
+            // # Click "Account settings"
+            cy.findByText('Account Settings').should('be.visible').click();
+
+            // * Check that the "Account Settings" modal was opened.
+            cy.get('#accountSettingsModal').should('exist').within(() => {
+                // # Click "Notifications"
+                cy.findByText('Notifications').should('be.visible').click();
+
+                // # Click "Desktop"
+                cy.findByText('Desktop Notifications').should('be.visible').click();
+
+                // # Select sound off.
+                cy.get('#soundOff').check();
+
+                // # Ensure sound dropdown is not visible
+                cy.get('#displaySoundNotification').should('not.be.visible');
+
+                // # Click "Save"
+                cy.findByText('Save').scrollIntoView().should('be.visible').click();
+
+                // Close the modal.
+                cy.get('#accountSettingsHeader').find('button').should('be.visible').click();
+            });
+
+            cy.apiGetChannelByName(testTeam.name, 'Off-Topic').then(({channel}) => {
+                const messageWithNotification = `random message with mention @${user.username}`;
+
+                // Have another user send a post with a mention
+                cy.postMessageAs({sender: testUser, message: messageWithNotification, channelId: channel.id});
+
+                // * Desktop notification is not received
+                cy.get('@withNotification').should('not.have.been.called');
```

Comment:
is it the same not having sound than not having notification?

although I think this one might be hard to test. notification api message should be set to silent, but that should still show the badge...

Reply:
Yes I was also a bit confused as well as the fact I don't get the notification displayed when testing locally